src/main/java/core/basesyntax/service/TransactionInvocationImpl.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax.service;
+
+import core.basesyntax.operations.TransactionExecutor;
+import java.util.List;
+import java.util.Map;
+
+public class TransactionInvocationImpl implements TransactionInvocation {
+    @Override
+    public void invokeTransaction(List<FruitTransaction> fruitTransactionList,
+                                  Map<FruitTransaction.Operation,
+                                          TransactionExecutor> operationsMap) {
+        for (FruitTransaction fruitTransaction : fruitTransactionList) {
+            new OperationStrategyImpl(operationsMap, fruitTransaction)
+                    .useOperation(fruitTransaction.getOperation());
```

Comment:
```suggestion
        for (FruitTransaction fruitTransaction : fruitTransactionList) {
            operationStrategy.get(fruitTransaction.getOperation())
                    .execute(fruitTransaction);
```
not need to create a few instances of OperationStrategy